presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -38,24 +38,22 @@ HttpClient::~HttpClient() {
 }
 
 HttpResponse::~HttpResponse() {
+  if (pool_ == nullptr) {
```

Comment:
We don't need this condition since pool_ is enforced upon construction. If it is null somehow (unlikely) we might want to let it fail in order to reveal issues early.